hcl/hcl_test.go

```diff
@@ -932,6 +1014,115 @@ func TestHCLParserStack(t *testing.T) {
 	}
 }
 
+func TestHCLParserMultipleErrors(t *testing.T) {
+	for _, tc := range []testcase{
+		{
+			name: "multiple syntax errors",
+			input: []cfgfile{
+				{
+					filename: "file.tm",
+					body:     "a=1\na=2\na=3",
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("file.tm", start(1, 1, 0), end(1, 2, 1))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file.tm", start(2, 1, 4), end(2, 2, 5))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file.tm", start(3, 1, 8), end(3, 2, 9))),
+				},
+			},
+		},
+		{
+			name: "multiple syntax errors in different files",
+			input: []cfgfile{
+				{
+					filename: "file1.tm",
+					body:     "a=1\na=2\na=3",
+				},
+				{
+					filename: "file2.tm",
+					body:     "a=1\na=2\na=3",
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("file1.tm", start(1, 1, 0), end(1, 2, 1))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file1.tm", start(2, 1, 4), end(2, 2, 5))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file1.tm", start(3, 1, 8), end(3, 2, 9))),
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("file2.tm", start(1, 1, 0), end(1, 2, 1))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file2.tm", start(2, 1, 4), end(2, 2, 5))),
+					errors.E(hcl.ErrHCLSyntax,
+						mkrange("file2.tm", start(3, 1, 8), end(3, 2, 9))),
+				},
+			},
+		},
+		{
+			name: "conflicting stack files",
+			input: []cfgfile{
+				{
+					filename: "stack1.tm",
+					body:     "stack {}",
+				},
+				{
+					filename: "stack2.tm",
+					body:     "stack {}",
+				},
+			},
+			want: want{
+				errs: []error{
```

Comment:
Conflicting git config produces two errors but here only stack2.tm is reported on the error list ? Is this expected ?

Reply:
the test called "conflicting terramate git config and other errors" has an additional error in it (the unknown `test` block) that's why it reports two errors.
Maybe I could separate this in another test case, but I added this to make sure conflicting errors still continue parsing the same file and generating other errors.

Reply:
Ah OK I misunderstood what was going on. All good :-)